python-spec/src/somacore/collection.py

```diff
@@ -20,34 +34,47 @@ class Collection(base.SOMAObject, MutableMapping[str, _ST], metaclass=abc.ABCMet
 
     __slots__ = ()
 
+    @classmethod
     @abc.abstractmethod
-    def add(
+    def create(
+        cls: Type[_CT],
+        uri: str,
+        *,
+        platform_config: Optional[options.PlatformConfig] = None,
+        context: Optional[Any] = None,
+    ) -> _CT:
+        """Creates a new Collection at the given URI and returns it.
+
+        The collection will be returned in the opened state.
+        """
+        raise NotImplementedError()
+
+    @abc.abstractmethod
+    def add_collection(
         self,
         key: str,
-        cls: Type[_ST],
+        cls: Optional[Type[_CT]] = None,
         *,
         uri: Optional[str] = None,
         platform_config: Optional[options.PlatformConfig] = None,
-    ) -> _ST:
-        """Creates a child member of this collection and adds it.
+    ) -> _CT:
+        """Creates a new sub-collection of this collection.
+
+        To add an existing collection as a sub-element of this collection,
+        use :meth:`set` or indexed access instead.
 
-        This allows the creation of child objects of any type::
+        The type provided is used to create the skeleton of this collection
+        as in :meth:`create`. By default, this will create a basic collection::
 
             # Create a child Measurement object at the key "density"
             # with default settings.
             #
             # This creates both the backing Measurement collection, as well as
             # the necessary sub-elements to have a complete Measurement.
-            density = the_collection.create("density", somacore.Measurement)
+            density = the_collection.add_collection("density", somacore.Measurement)
 
-            # Create a child DataFrame at the key "data" with a custom URI
-            # and additional platform configuration.
-            data = the_collection.create(
-                "data",
-                somacore.DataFrame,
-                uri="file:///custom/path/to/df",
-                platform_config={"somaimpl": ...},
-            )
+            # This will create a basic Collection as a child.
+            sub_child = density.add_collection("sub_child")
 
         By default (in situations where it is possible to do so), the default
```

Comment:
is this a suggestion/best practice, or a requirement?  Some keys are not legal URI path components (eg, may contain characters illegal in a URI).  If just a best practice, perhaps label it as such.

Reply:
I've added more details about what should be done here:

- Use the key as the sub-URI where possible
- Sanitize it if need be